### PR TITLE
EOS-18945: Do not restart service for miniprovisioner sspl_test.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
## Problem Statement
<pre>
  <code>
    Story Ref (if any): https://jts.seagate.com/browse/EOS-18945
    Avoid service restart for self -test.
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Do not restart service for miniprovisioner sspl-test ie self_test 
  </code>
</pre>
## Solution
<pre>
  <code>
    Please add summary of the change,List any dependencies that are required for this change.
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes

  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py [hunk 1]
```diff
@@ -63,90 +63,99 @@ def validate():
...
+            Conf.set(SSPL_CONFIG_INDEX,"SERVICEMONITOR>monitored_services",
+                service_list)
+            Conf.set(SSPL_CONFIG_INDEX, "SERVICEMONITOR>threshold_inactive_time",
+                threshold_inactive_time_original)
+            Conf.set(SSPL_CONFIG_INDEX,
+                    "SYSTEM_INFORMATION>global_config_copy_url", global_config_copy_url)
+            Conf.save(SSPL_CONFIG_INDEX)
+            shutil.copyfile(sspl_test_backup, sspl_test_file_path)
+
+            self.dbus_service.restart('sspl-ll.service')
```
Comment (mariyappanp) [inline low-level/files/opt/seagate/sspl/setup/sspl_test.py:153]: Please remove this restart and verify once again.

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py [hunk 2]
```diff
@@ -63,90 +63,99 @@ def validate():
...
+            if returncode != 0:
+                raise SetupError(returncode, "%s - ERROR: %s - CMD %s", self.name, error, CMD)
+        else:
+            # TODO: Convert shell script to python
+            # from cortx.sspl.sspl_test.run_qa_test import RunQATest
+            # RunQATest(self.plan, self.avoid_rmq).run()
+            CMD = "%s/run_qa_test.sh %s %s" % (TEST_DIR, self.plan, self.avoid_rmq)
```
Comment (mariyappanp) [inline low-level/files/opt/seagate/sspl/setup/sspl_test.py:160]: There is a chance for KeyboardInterrupt error if user forcefully exit the test.
If that happens, please make sure test related config changes are reverted.

**Review by cdeshmukh — APPROVED** [review]
Address open review comments and am good to merge

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py (hunk 1 above)
Comment (shriya-deshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py]: Tried sanity after removing restart, sanity will run without any issue,
But observed one thing ie, For sanity test, after test completion  we restored changed config to original config, (in run_qa_test.sh script-restored_config).
So if we remove this restart then that restored config will not reflect in service, Service will still used test config.

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py (hunk 2 above)
Comment (shriya-deshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py:175]: Done.

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py (hunk 1 above)
Comment (mariyappanp) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py]: I have removed this in PR #347 
Please rebase your repo and check this scenario.

**Review by cdeshmukh — COMMENTED** [review]
@thavanathan @sandeepsgt please review this

**Review by sandeepsgt — APPROVED** [review]
Test name can be changed, instead of `self_primary` and  `self_secondary`, we can just have `self`

**Comment by shriya-deshmukh** [comment]
>> Test name can be changed, instead of self_primary and self_secondary, we can just have self

added "self.pln" in test plans, should i remove self_primary and self_secondary.pln files ?

File: sspl_test/plans/self.pln [hunk 3]
```diff
@@ -0,0 +1,4 @@
+alerts.self_hw.test_self_hw_real_stor_enclosure_conn
```
Comment (mariyappanp) [inline sspl_test/plans/self.pln:1]: Ensure that node_id that from `from alerts.self_hw.self_hw_utilities import get_node_id` is only from GLOBAL_CONF.
There it is hard code to 'srvnode-1'.

Comment (mariyappanp) [reply sspl_test/plans/self.pln:1]: Note: Also, we are advised to avoid direct salt commands.

Comment (shriya-deshmukh) [reply sspl_test/plans/self.pln:1]: Done, Instead of salt cmd, used  conf store api to read node_key value , in `"from alerts.self_hw.self_hw_utilities import get_node_id".`

**Review by mariyappanp — APPROVED** [review]

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py [hunk 4]
```diff
@@ -76,85 +76,96 @@ def process(self):
...
+            sspl_global_config_url = Conf.get(SSPL_CONFIG_INDEX,
+                "SYSTEM_INFORMATION>global_config_copy_url")
+            Conf.set(SSPL_CONFIG_INDEX, "SYSTEM_INFORMATION>global_config_copy_url",
+                self.sspl_test_gc_copy_url)
+            Conf.save(SSPL_CONFIG_INDEX)
+
+            # Enable & disable sensors based on environment
+            update_sensor_info(SSPL_TEST_CONFIG_INDEX, self.node_type, self.enclosure_type)
+
+            # TODO: Move lines 90-116 & 125-127 to RunQATest class
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_test.py:97]: these line numbers doesnt match anymore i believe. See if needed still, or cleanup his TODO line

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py [hunk 5]
```diff
@@ -76,85 +76,96 @@ def process(self):
...
+                raise Exception("%s - ERROR: %s - CMD %s" % (self.name, error, CMD))
+        else:
+            # TODO: Convert shell script to python
+            # from cortx.sspl.sspl_test.run_qa_test import RunQATest
+            # RunQATest(self.plan, self.avoid_rmq).run()
+            try:
+                CMD = "%s/run_qa_test.sh %s %s" % (TEST_DIR, self.plan, self.avoid_rmq)
+                output, error, returncode = SimpleProcess(CMD).run(realtime_output=True)
+            except KeyboardInterrupt:
+                print("KeyboardInterrupt occurred while executing sspl test.")
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_test.py:171]: generic exception also in sspl_setup (or somewhere) to catch all exceptions, and print rc?

File: low-level/sensors/impl/generic/node_hw.py [hunk 6]
```diff
@@ -260,7 +260,6 @@ def initialize(self, conf_reader, msgQlist, products):
             ServiceTypes.SERVER_NODE.value)
         self.__bmc_passwd = encryptor.decrypt(decryption_key,
             _bmc_secret, self.SENSOR_NAME)
-
```
Comment (cdeshmukh) [inline low-level/sensors/impl/generic/node_hw.py:263]: if no change, avoid this file from changed list

File: sspl_test/alerts/self_hw/self_hw_utilities.py [hunk 7]
```diff
@@ -37,24 +38,9 @@ def is_virtual():
...
-                result = subout.stdout.readlines()
-                if result == [] or result == "":
-                    print(f"Command '{MINION_GET_NODE_CMD}' failed to fetch minion id or hostname.")
-                    print("Using node_id ('srvnode-1') as we are not able to fetch it from hostname command.")
-                    node_key = 'srvnode-1'
-                else:
-                    node_key = result[0].decode().rstrip('\n').lower()
-            else:
-                node_key = result[0].decode().rstrip('\n')
+            node_key = Conf.get(GLOBAL_CONF, NODE_KEY)
```
Comment (cdeshmukh) [inline sspl_test/alerts/self_hw/self_hw_utilities.py:43]: collection should happen in node_id. I know its existing code, lets change that too now

File: sspl_test/plans/self_primary.pln [hunk 8]
```diff
@@ -1,4 +1,4 @@
 alerts.self_hw.test_self_hw_real_stor_enclosure_conn
 alerts.node.test_node_fan_actuator
 alerts.self_hw.test_self_hw_node_sel_event
-alerts.realstor.test_real_stor_fan_actuator
\ No newline at end of file
+alerts.realstor.test_real_stor_fan_actuator
```
Comment (cdeshmukh) [inline sspl_test/plans/self_primary.pln:4]: no change here, file should not reflect in PR

File: sspl_test/plans/self_secondary.pln [hunk 9]
```diff
@@ -1,3 +1,3 @@
 alerts.self_hw.test_self_hw_real_stor_enclosure_conn
 alerts.node.test_node_fan_actuator
-alerts.self_hw.test_self_hw_node_sel_event
\ No newline at end of file
+alerts.self_hw.test_self_hw_node_sel_event
```
Comment (cdeshmukh) [inline sspl_test/plans/self_secondary.pln:3]: no change here, file should not reflect in PR

File: sspl_test/run_qa_test.sh [hunk 10]
```diff
@@ -315,10 +315,10 @@ if [ "$IS_VIRTUAL" == "true" ]
 then
     # setting back the actual values
     $sudo $script_dir/set_threshold.sh $transmit_interval $disk_usage_threshold $host_memory_usage_threshold $cpu_usage_threshold $sspl_config
-    [[ -f ${SSPL_CONFIG_FILE}.back ]] && $sudo mv ${SSPL_CONFIG_FILE}.back $SSPL_CONFIG_FILE
-    #[[ -f ${test_config_file}.back ]] && $sudo mv ${test_config_file}.back $test_config_file
-    [[ -f ${global_config_file}.back ]] && $sudo mv ${global_config_file}.back $global_config_file
 fi
+[[ -f ${SSPL_CONFIG_FILE}.back ]] && $sudo mv ${SSPL_CONFIG_FILE}.back $SSPL_CONFIG_FILE
```
Comment (cdeshmukh) [inline sspl_test/run_qa_test.sh:319]: when config not being changed for 'self' test plan, why these changes are being brought out, for control flow beyond virtual env ?

Comment (shriya-deshmukh) [reply sspl_test/run_qa_test.sh:319]: >> for control flow beyond virtual env ?
yes and also
at line 171 and 173 we are creating backup, there we don't have check 'if [ "$IS_VIRTUAL" == "true" ]',  hence these lines brought out of virtual env check.

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py (hunk 4 above)
Comment (shriya-deshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py]: Updated line numbers.

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py (hunk 5 above)
Comment (shriya-deshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py]: Done. defined TestException class in same file, to catch all the exception.

File: sspl_test/alerts/self_hw/self_hw_utilities.py (hunk 7 above)
Comment (shriya-deshmukh) [reply sspl_test/alerts/self_hw/self_hw_utilities.py]: done.

File: low-level/sensors/impl/generic/node_hw.py (hunk 6 above)
Comment (shriya-deshmukh) [reply low-level/sensors/impl/generic/node_hw.py]: removed.

File: sspl_test/plans/self_secondary.pln (hunk 9 above)
Comment (shriya-deshmukh) [reply sspl_test/plans/self_secondary.pln]: removed.

File: sspl_test/plans/self_primary.pln (hunk 8 above)
Comment (shriya-deshmukh) [reply sspl_test/plans/self_primary.pln]: Done.

**Review by cdeshmukh — APPROVED** [review]
